test/api.test.js

```diff
@@ -0,0 +1,331 @@
+
+          pull(
+            sbot.metafeeds.branchStream({
+              root: indexesMF.subfeed,
+              old: true,
+              live: false,
+            }),
+            pull.collect((err, branches) => {
+              if (err) reject(err)
+
+              t.equal(branches.length, 2, '2 branches')
+
+              t.equal(branches[0].length, 1, 'indexes branch')
+              t.equal(
+                branches[0][0][1].feedpurpose,
+                'indexes',
+                'indexes branch'
+              )
+
+              t.equal(branches[1].length, 2, 'index branch')
+              t.equal(branches[1][1][1].feedpurpose, 'about', 'indexes branch')
+            })
+          )
+
+          resolve()
+        })
+      )
+    })
+  }
+
+  await testBranchStream(sbot).catch(t.error)
+
+  console.log('persistence')
+  await p(sbot.close)()
+  sbot = Testbot({ path, rimraf: false })
+  await testBranchStream(sbot).catch(t.error)
+
+  sbot.close()
+  t.end()
+})
+
+test('findAndTombstone and tombstoning branchStream', async (t) => {
+  let sbot = Testbot()
+  const { path } = sbot.config
+
+  const { rootMF } = await setupTree(sbot)
+
+  let todo = 3
+  pull(
+    sbot.metafeeds.branchStream({ tombstoned: true, old: false, live: true }),
+    pull.drain((branch) => {
+      t.equals(branch.length, 2, 'live')
+      t.equals(branch[0][0], rootMF.keys.id, 'live')
+      t.equals(branch[1][1].feedpurpose, 'chess', 'live')
+      t.equals(branch[1][1].reason, 'This game is too good', 'live')
+
+      testOldStreams(sbot)
+    })
+  )
+
+  function testOldStreams(sbot) {
+    pull(
+      sbot.metafeeds.branchStream({ tombstoned: true, old: true, live: false }),
+      pull.drain((branch) => {
+        t.equals(branch.length, 2)
+        t.equals(branch[0][0], rootMF.keys.id, 'tombstoned: true')
+        t.equals(branch[1][1].feedpurpose, 'chess', 'tombstoned: true')
+        t.equals(
+          branch[1][1].reason,
+          'This game is too good',
+          'tombstoned: true'
+        )
+        todo--
+      })
+    )
+
+    pull(
+      sbot.metafeeds.branchStream({
+        tombstoned: false,
+        old: true,
+        live: false,
+      }),
+      pull.collect((err, branches) => {
+        if (err) throw err
+        t.equal(branches.length, 4, 'tombstoned: false')
+        todo--
+      })
+    )
+
+    pull(
+      sbot.metafeeds.branchStream({ tombstoned: null, old: true, live: false }),
+      pull.collect((err, branches) => {
+        if (err) throw err
+        t.equal(branches.length, 5, 'tombstoned: null')
+        todo--
+      })
+    )
+  }
+
+  await p(sbot.metafeeds.findAndTombstone)(
+    rootMF,
+    (f) => f.feedpurpose === 'chess',
+    'This game is too good'
+  )
+
+  while (todo > 0) await p(setTimeout)(100)
+
+  console.log('(persistence)')
+  await p(sbot.close)()
+  sbot = Testbot({ path, rimraf: false })
+
+  todo = 3
+  testOldStreams(sbot)
```

Comment:
these test fail after sbot restart.... not sure why. Output suggested the branches have not been loaded correctly.
That or I'm expecting different behaviour than you!

![image](https://user-images.githubusercontent.com/2665886/191171878-d066668d-069d-41e5-9d0d-55b78092d296.png)

Reply:
Why was this test rewritten with promises? It seems like you could have just used the new `Testbot()` helper and kept all the rest the same as before. With less things changing, you're more certain as to why things are breaking.

So my suggestion is to bring back the test as it was before, but just using Testbot() this time.

Reply:
https://github.com/ssbc/ssb-meta-feeds/pull/74